plugins/Live/javascripts/visitorProfile.js

```diff
@@ -44,27 +44,46 @@
      * @param {String} idSite The ID of the site.
      */
     VisitorProfileControl.showPopover = function (visitorId, idSite) {
-
-        if (!piwik.visitorProfileEnabled) {
+        var showProfile = function(visitorId, idSite, visitorProfileEnabled) {
+          if (!piwik.visitorProfileEnabled && (!idSite || piwik.idSite == idSite)) {
             console.error('Visitor Profile was disabled in website settings');
             return;
-        }
+          }
 
-        var url = 'module=Live&action=getVisitorProfilePopup&visitorId=' + encodeURIComponent(visitorId);
-        if (idSite) {
+          var url = 'module=Live&action=getVisitorProfilePopup&visitorId=' + encodeURIComponent(visitorId);
+          if (idSite) {
             url += '&idSite=' + idSite;
-        }
+          }
 
-        // if there is already a map shown on the screen, do not show the map in the popup. kartograph seems
-        // to only support showing one map at a time.
-        if ($('.RealTimeMap').length > 0) {
+          // if there is already a map shown on the screen, do not show the map in the popup. kartograph seems
+          // to only support showing one map at a time.
+          if ($('.RealTimeMap').length > 0) {
             url += '&showMap=0';
+          }
+
+          var ajaxRequest = new ajaxHelper();
+          ajaxRequest.removeDefaultParameter('segment');
+
+          Piwik_Popover.createPopupAndLoadUrl(url, _pk_translate('Live_VisitorProfile'), 'visitor-profile-popup', ajaxRequest);
+        }
+
+        if (idSite && idSite != piwik.idSite) {
+          var ajax = new ajaxHelper();
+          ajax.addParams({
+            module: 'API',
+            method: 'Live.isVisitorProfileEnabled',
+            format: 'json',
+            idSite: idSite
+          }, 'GET');
+          ajax.setCallback(function (response) {
+            showProfile(visitorId, idSite, response && response.value);
+          });
+          ajax.send();
+        } else {
+          showProfile(visitorId, idSite, piwik.visitorProfileEnabled);
```

Comment:
This change ensure that the visitor profile will open for a site where it is enabled. The previous check was only for the current site, which is incorrect on GDPR tools. Now it will first check if the profile is enabled before triggering the profile. There might be better solutions and we could cache that somehow maybe, but might be good enough for now.